hack/update-codecgen.sh

```diff
@@ -41,6 +41,52 @@ function cleanup {
 }
 trap cleanup EXIT
 
+# Sort all files in the dependency order.
+number=${#generated_files[@]}
+for (( i=0; i<${number}; i++ )); do
+  visited[${i}]=false
+done
+result=""
+
+function fullpath {
```

Comment:
Does `readlink -f` not work? (http://stackoverflow.com/questions/17577093/how-do-i-get-the-absolute-directory-of-a-file-in-bash)